numba/cuda/tests/cudadrv/test_cuda_ndarray.py

```diff
@@ -58,6 +58,29 @@ def test_devicearray_replace(self):
         gpumem.copy_to_host(array)
         self.assertTrue((array == original * 2).all())
 
+    def test_devicearray_transpose_wrongdim(self):
+        gpumem = cuda.to_device(np.array(np.arange(12)).reshape(3, 4, 1))
+        self.assertRaises(NotImplementedError, lambda: np.transpose(gpumem))
+
+    def test_devicearray_transpose_identity(self):
+        original = np.array(np.arange(12)).reshape(3, 4)
+        array = np.transpose(cuda.to_device(original), axes=(0, 1)).copy_to_host()
+        self.assertTrue(np.all(array == original))
+
+    def test_devicearray_transpose_duplicatedaxis(self):
+        gpumem = cuda.to_device(np.array(np.arange(12)).reshape(3, 4))
+        self.assertRaises(ValueError, lambda: np.transpose(gpumem, axes=(0, 0)))
```

Comment:
It'll be better to check the message in the exception to make sure that the `ValueError` is raised by the check in transpose.